commerce-service/src/main/java/com/liferay/commerce/internal/messaging/CommerceOrderStatusMessageListener.java

```diff
@@ -44,17 +53,46 @@
 	protected void doReceive(Message message) throws Exception {
 		int orderStatus = message.getInteger("orderStatus");
 
-		if (orderStatus == CommerceOrderConstants.ORDER_STATUS_TO_TRANSMIT) {
-			long commerceOrderId = message.getLong("commerceOrderId");
+		long commerceOrderId = message.getLong("commerceOrderId");
 
-			CommerceOrder commerceOrder =
-				_commerceOrderLocalService.getCommerceOrder(commerceOrderId);
+		CommerceOrder commerceOrder =
+			_commerceOrderLocalService.getCommerceOrder(commerceOrderId);
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannelByOrderGroupId(
+				commerceOrder.getGroupId());
+
+		boolean hasWorkflow =
+			_workflowDefinitionLinkLocalService.hasWorkflowDefinitionLink(
+				commerceOrder.getCompanyId(), commerceOrder.getGroupId(),
+				CommerceOrder.class.getName(), 0,
+				CommerceOrderConstants.TYPE_PK_APPROVAL);
+
+		if (hasWorkflow &&
+			(orderStatus == CommerceOrderConstants.ORDER_STATUS_OPEN) &&
+			(commerceOrder.getStatus() != WorkflowConstants.STATUS_APPROVED)) {
 
 			// Commerce notifications
 
-			CommerceChannel commerceChannel =
-				_commerceChannelLocalService.getCommerceChannelByOrderGroupId(
-					commerceOrder.getGroupId());
+			_commerceNotificationHelper.sendNotifications(
+				commerceChannel.getSiteGroupId(),
+				OrderSubmittedCommerceNotificationTypeImpl.KEY, commerceOrder);
```

Comment:
@luca-pellizzon using the `KEY`s like this, it would make more sense to me for them to be in an internal constants file, rather than a `KEY` field in each separate impl. What do you think?

Reply:
It _might_ even make sense to add `ORDER_LABEL_*` constants to `CommerceOrderConstants` (rather than just listing them as string literals in `getOrderStatusLabel`), and then use those `ORDER_LABEL`s for the `KEY`s. What would you think of that? :)

Reply:
Not sure to understand this comment. The KEYS are referring to the notification type. They are used to identify the correct notification from the 'CommerceNotificationTypeRegistry'

Reply:
Sorry, @luca-pellizzon , it just seems like it would be nice to have all the `KEY` constants gathered together into a single Constants file and then referenced from there. Because it seems like each one represents a status. And it would be especially cool if we could reuse some existing constants to do that (but this may not make any sense). If you don't think any of this makes any sense, and you want to just keep them as `KEY`s, that's fine, it's not a big deal -- just seems like it might make things a little more obvious.

Reply:
Thanks @ethanbustad now it's clear.
Let me check what I can do